source/union-to-intersection.d.ts

```diff
@@ -0,0 +1,59 @@
+/**
+A helpful utility which maps a union type to an intersection type using [distributive conditional types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#distributive-conditional-types).
+
+Inspired by a [stack overflow answer](https://stackoverflow.com/a/50375286/2172153).
+
+@example
+
+```ts
+import {UnionToIntersection} from 'type-fest';
+
+type Union = {the(): void} | {great(arg: string): void} | {escape: boolean};
+type Intersection = UnionToIntersection<Union>;
+//=> {the(): void; great(arg: string): void; escape: boolean};
+```
+
+A more applicable example which could make it's way into your library code follows.
+
+@example
+
+```ts
+import {UnionToIntersection} from 'type-fest';
+
+class CommandOne {
+  commands: {
+    a1: () => undefined,
+    b1: () => undefined,
+  }
+}
+
+class CommandTwo {
+  commands: {
+    a2: (argA: string) => undefined,
+    b2: (argB: string) => undefined,
+  }
+}
+
+const union = [new CommandOne(), new CommandTwo()].map(instance => instance.commands)
+type Union = typeof union;
+//=> {a1(): void; b1(): void} | {a2(argA: string): void; b2(argB: string): void}
+
+type Intersection = UnionToIntersection<Union>;
+//=> {a1(): void; b1(): void; a2(argA: string): void; b2(argB: string): void}
+```
+*/
+export type UnionToIntersection<Union> = (
+  // `extends any` is always going to be the case and is used to convert the
+  // `Union` into a [distributive conditional
+  // type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#distributive-conditional-types).
+  Union extends any
```

Comment:
Can you use `unknown` instead of `any`?

Reply:
Who would have thought 😄 

Nice catch!